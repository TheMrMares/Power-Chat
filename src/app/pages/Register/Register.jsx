import React from 'react'
import PropTypes from 'prop-types'
import {
  Paper,
  withStyles,
} from '@material-ui/core'
import styles from './Register.style'

export class Register extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
  }

  render() {
    const { classes } = this.props
    return (
      <div className={classes.root}>
        <Paper className={classes.paper}>
          REgister
        </Paper>
      </div>
    )
  }
}

export default withStyles(styles)(Register)
