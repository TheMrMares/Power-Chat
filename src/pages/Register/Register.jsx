import React from 'react'
import { Link } from 'react-router-dom'
import Button from '../../components/Button'
import Input from '../../components/Input'
import { isEmailValid } from '../../utils/formValidation'
import styles from './Register.module.scss'

export class Register extends React.Component {
  constructor() {
    super()
    this.state = {
      registerEmail: { data: '', error: '' },
      registerFirstname: { data: '', error: '' },
      registerNickname: { data: '', error: '' },
      registerPassword: { data: '', error: '' },
      registerRepeatEmail: { data: '', error: '' },
      registerRepeatPassword: { data: '', error: '' },
      registerSurname: { data: '', error: '' },
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  async handleSubmit(evt) {
    evt.preventDefault()
    evt.stopPropagation()
    const { state } = this
    let overallCondition = false
    if (!isEmailValid(state.registerEmail.data)) {
      this.setState({
        registerEmail: {
          data: state.registerEmail.data,
          error: 'Your email is incorrect',
        },
      })
      overallCondition = true
    }
    if (state.registerRepeatEmail.data !== state.registerEmail.data) {
      this.setState({
        registerRepeatEmail: {
          data: state.registerRepeatEmail.data,
          error: 'Repeated email have to be same',
        },
      })
      overallCondition = true
    }
    if (!(state.registerNickname.data.length >= 6 && state.registerNickname.data.length <= 16)) {
      this.setState({
        registerNickname: {
          data: state.registerNickname.data,
          error: 'Nickname have to be min. 6 and max. 16 characters',
        },
      })
      overallCondition = true
    }
    if (state.registerFirstname.data === '') {
      this.setState({
        registerFirstname: {
          data: state.registerFirstname.data,
          error: 'Firstname cannot be empty',
        },
      })
      overallCondition = true
    }
    if (state.registerSurname.data === '') {
      this.setState({
        registerSurname: {
          data: state.registerSurname.data,
          error: 'Surname cannot be empty',
        },
      })
      overallCondition = true
    }
    if (!(state.registerPassword.data.length >= 6 && state.registerPassword.data.length <= 24)) {
      this.setState({
        registerPassword: {
          data: state.registerPassword.data,
          error: 'Password have to be min. 6 and max. 16 characters',
        },
      })
      overallCondition = true
    }
    if (state.registerPassword.data !== state.registerRepeatPassword.data) {
      this.setState({
        registerRepeatPassword: {
          data: state.registerRepeatPassword.data,
          error: 'Passwords must be same',
        },
      })
      overallCondition = true
    }

    if (overallCondition === true) {
      return
    }
  }

  handleChange(evt) {
    const { id, value } = evt.target
    this.setState({
      [id]: {
        data: value,
        error: '',
      },
    })
  }

  render() {
    const { state } = this
    return (
      <div className={styles.wrapper}>
        <form className={styles.form}>
          <h1>Register</h1>
          <div className={styles.row}>
            <div className={styles.column}>
              <h2>Email</h2>
              <Input
                error={state.registerEmail.error}
                id="registerEmail"
                label="Email"
                onChange={this.handleChange}
                placeholder="example@domain.com"
                type="email"
                value={state.registerEmail.data}
              />
              <Input
                error={state.registerRepeatEmail.error}
                id="registerRepeatEmail"
                label="Repeat email"
                onChange={this.handleChange}
                placeholder="example@domain.com"
                type="email"
                value={state.registerRepeatEmail.data}
              />
              <h2>User</h2>
              <Input
                error={state.registerNickname.error}
                id="registerNickname"
                label="Nickname"
                onChange={this.handleChange}
                placeholder="HappyNickname123"
                type="text"
                value={state.registerNickname.data}
              />
            </div>
            <div className={styles.column}>
              <h2>Personal</h2>
              <Input
                error={state.registerFirstname.error}
                id="registerFirstname"
                label="Firstname"
                onChange={this.handleChange}
                placeholder="John"
                type="text"
                value={state.registerFirstname.value}
              />
              <Input
                error={state.registerSurname.error}
                id="registerSurname"
                label="Surname"
                onChange={this.handleChange}
                placeholder="Smith"
                type="text"
                value={state.registerSurname.value}
              />
            </div>
            <div className={styles.column}>
              <h2>Password</h2>
              <Input
                error={state.registerPassword.error}
                id="registerPassword"
                label="Choose password"
                onChange={this.handleChange}
                placeholder="******"
                type="password"
                value={state.registerPassword.data}
              />
              <Input
                error={state.registerRepeatPassword.error}
                id="registerRepeatPassword"
                label="Repeat password"
                onChange={this.handleChange}
                placeholder="******"
                type="password"
                value={state.registerRepeatPassword.data}
              />
            </div>
          </div>
          <Button
            onClick={this.handleSubmit}
            size="normal"
            uppercase
          >
            register
          </Button>
          <p>Have you already got an account?</p>
          <Link to="/login">Sign in.</Link>
        </form>
      </div>
    )
  }
}

export default Register
