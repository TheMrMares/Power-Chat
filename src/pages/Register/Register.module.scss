@import '../../css/config.scss';
@import '../../css/mixins.scss';

.wrapper {
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: $spacing-unit * 1;
}
.form {
  background: $primary-normal;
  padding: $spacing-unit;
  border-radius: $spacing-radius;
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
  padding: $spacing-unit*2;
  align-items: center;
  border-left: 6px solid;
  border-right: 6px solid;
  border-color: $primary-light;
  @include respond-to(small) {
    flex: none;
    min-width: 45%;
  }
  @include respond-to(medium) {
    margin: 0;
    min-width: 35%;
  }
  @include respond-to(large) {
    min-width: 25%;
  }
}

.row {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  @include respond-to(medium) {
    flex-direction: row;
  }
  .column {
    display: flex;
    flex-direction: column;
    align-items: stretch;
    @include respond-to(medium) {
      align-items: flex-start;
    }
  }
}
