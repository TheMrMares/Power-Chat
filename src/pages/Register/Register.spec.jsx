import React from 'react'
import { shallow } from 'enzyme'
import { Register } from './Register'

const initPage = overrides => {
  const wrapper = shallow(
    <Register
      {...overrides}
    />
  )
  return { wrapper }
}

describe('Register', () => {
  it('should render as expected', () => {
    const { wrapper } = initPage()
    expect(wrapper).toMatchSnapshot()
  })
})
