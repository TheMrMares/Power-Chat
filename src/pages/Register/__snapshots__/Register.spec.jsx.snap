// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render as expected 1`] = `
<div>
  <form>
    <h1>
      Register
    </h1>
    <div>
      <div>
        <h2>
          Email
        </h2>
        <Input
          error=""
          id="registerEmail"
          label="Email"
          onChange={[Function]}
          placeholder="example@domain.com"
          type="email"
          value=""
        />
        <Input
          error=""
          id="registerRepeatEmail"
          label="Repeat email"
          onChange={[Function]}
          placeholder="example@domain.com"
          type="email"
          value=""
        />
        <h2>
          User
        </h2>
        <Input
          error=""
          id="registerNickname"
          label="Nickname"
          onChange={[Function]}
          placeholder="HappyNickname123"
          type="text"
          value=""
        />
      </div>
      <div>
        <h2>
          Personal
        </h2>
        <Input
          error=""
          id="registerFirstname"
          label="Firstname"
          onChange={[Function]}
          placeholder="John"
          type="text"
        />
        <Input
          error=""
          id="registerSurname"
          label="Surname"
          onChange={[Function]}
          placeholder="Smith"
          type="text"
        />
      </div>
      <div>
        <h2>
          Password
        </h2>
        <Input
          error=""
          id="registerPassword"
          label="Choose password"
          onChange={[Function]}
          placeholder="******"
          type="password"
          value=""
        />
        <Input
          error=""
          id="registerRepeatPassword"
          label="Repeat password"
          onChange={[Function]}
          placeholder="******"
          type="password"
          value=""
        />
      </div>
    </div>
    <Button
      onClick={[Function]}
      size="normal"
      uppercase={true}
    >
      register
    </Button>
    <p>
      Have you already got an account?
    </p>
    <Link
      replace={false}
      to="/login"
    >
      Sign in.
    </Link>
  </form>
</div>
`;
